src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -50,6 +85,29 @@ export default class ReplyField extends React.Component<Props, State> {
         this.saveCursorPosition();
     }
 
+    getContainerRect = (): DOMRect | null => {
+        const { current: container } = this.containerRef;
+
+        return container ? container.getClientRects().item(0) : null;
+    };
+
+    handlePopupChange = (): void => {
+        const { containerRect: prevRect } = this.state;
+        const rect = this.getContainerRect();
+        const isSame = prevRect?.x === rect?.x && prevRect?.y === rect?.y;
+
+        if (!isSame) {
+            this.updateRectAndPopup(rect);
+        }
+    };
+
+    updateRectAndPopup = (rect: DOMRect | null): void => {
+        this.setState({
+            containerRect: rect,
```

Comment:
We should be able to derive the container rect at any time based on the rect. This seems like duplicated state. Why is this needed?

Reply:
> We should be able to derive the container rect at any time based on the rect. This seems like duplicated state. Why is this needed?

I need to get previous rect. If they're not the same, I need to re-render popupList to follow the container new position.